compiler/rustc_target/src/asm/x86.rs

```diff
@@ -152,13 +152,41 @@ fn high_byte(
     }
 }
 
+fn rbx_reserved(
+    arch: InlineAsmArch,
+    _has_feature: impl FnMut(&str) -> bool,
+    _target: &Target,
+) -> Result<(), &'static str> {
+    match arch {
+        InlineAsmArch::X86 => Ok(()),
+        InlineAsmArch::X86_64 => {
+            Err("rbx is used internally by LLVM and cannot be used as an operand for inline asm")
```

Comment:
From the RFC:

> The intent is that support for asm! should be independent of the rustc back-end used

Given that, I think backend limitations such as these should be handled by rustc prior to passing the assembly to the backend. It shouldn't be visible to the user. Otherwise, `asm!` would only be able to support the intersection of all inline assembly implementations across all codegen backends.

Reply:
Handling this in rustc is actually non-trivial (see [this long Zulip discussion](https://rust-lang.zulipchat.com/#narrow/stream/216763-project-inline-asm/topic/Handling.20clobbers)), which is also why LLVM doesn't just do it for us.